app/Filament/Resources/ServerResource/Pages/ListServers.php

```diff
@@ -64,9 +64,8 @@ public function table(Table $table): Table
                     ->searchable(),
                 SelectColumn::make('allocation_id')
                     ->label('Primary Allocation')
-                    ->options(fn (Server $server) => $server->allocations->mapWithKeys(
-                        fn ($allocation) => [$allocation->id => $allocation->address])
-                    )
+                    ->disabled(fn () => !auth()->user()->can('update server'))
+                    ->options(fn (Server $server) => $server->allocations->mapWithKeys(fn ($allocation) => [$allocation->id => $allocation->address]))
```

Comment:
Instead of disabling the select itself it would probably look nicer to hide the column and show a normal TextColumn instead.

Reply:
Something like that ?
```php
auth()->user()->can('update server')
                    ? SelectColumn::make('allocation_id')
                        ->label('Primary Allocation')
                        ->options(fn (Server $server) => $server->allocations->mapWithKeys(fn ($allocation) => [$allocation->id => $allocation->address]))
                        ->selectablePlaceholder(false)
                        ->sortable()
                    : TextColumn::make('allocation_id')
                        ->label('Primary Allocation')
                        ->formatStateUsing(fn (Server $server) => $server->allocation->address)
                        ->sortable(),
```

Reply:
> Something like that ?
> 
> ```
> auth()->user()->can('update server')
>                     ? SelectColumn::make('allocation_id')
>                         ->label('Primary Allocation')
>                         ->options(fn (Server $server) => $server->allocations->mapWithKeys(fn ($allocation) => [$allocation->id => $allocation->address]))
>                         ->selectablePlaceholder(false)
>                         ->sortable()
>                     : TextColumn::make('allocation_id')
>                         ->label('Primary Allocation')
>                         ->formatStateUsing(fn (Server $server) => $server->allocation->address)
>                         ->sortable(),
> ```

Yes but controlled with `hidden`/ `visible`.

Reply:
I doubt we can have multiple fields with the same name just using `hidden` / `visible` which is why i went with ternary

Reply:
I should work since only of them is actually visible at a time. But let's try and see.

Reply:
Well after testing i see that its procedural so the first one got overwritten.
With permission it shows TextColumn and without it just removes that column

Reply:
Rename the `TextColimn` to something like `allocation_id_readonly`. It's using `formatStateUsing` to overwrite the state anyways so it should be fine.